api/app/Console/Commands/CallFactory.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Exception;
+use Illuminate\Console\Command;
+
+class CallFactory extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'app:call-factory
+                            {class : The factory class to call}
+                            {--state= : The factory state to use}
+                            {--attributes= : Override attributes to pass to the factory (JSON)}
+                            ';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Calls an Eloquent factory to instantiate a model directly';
+
+    /**
+     * Execute the console command.
+     */
+    public function handle()
+    {
+        $result = null;
+
+        $qualifiedClassName = 'App\\Models\\' . $this->argument("class");
+        if (!class_exists($qualifiedClassName)) {
+            throw new Exception("The model class does not exist.");
+        }
+        if (!method_exists($qualifiedClassName, "factory")) {
+            throw new Exception("The model class does not have a factory method.");
+        }
+
+        $jsonAttributes = $this->option('attributes');
+        if (!empty($jsonAttributes)) {
+            $attributes = json_decode($jsonAttributes, true);
+        } else {
+            $attributes = null;
+        }
+
+        $state = $this->option('state');
+        if (empty($state)) {
+            $result = $qualifiedClassName::factory()->create($attributes);
+        } else {
+            $result = $qualifiedClassName::factory()->$state()->create($attributes);
```

Comment:
I'm wondering, could we make state accept more than one item?

> **Note**
> I have no idea if this will work 😛 

```php
$state = $this->option('state');
if(empty($state)) {
  ...
} else {
  $states = explode(",", $state);
  $factoryWithState = $qualifiedClassName::factory();
  foreach($states as $stateMethod) {
     $factoryWithState = $factoryWithState->$stateMethod();
  }
  $result = $factoryWithState->create($attributes);
}
```

So then I could call:

```sh
php artisan app:call-factory User --state=asApplicant,withExperiences,withSkills
```

Reply:
Artisan commands accept arguments with more than one value natively so we wouldn't have to do the explode.  I'll try it out!

Reply:
Good because I don't like that function name 😝

Reply:
Cool, it works!
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/329c1252-e6cf-4163-b21a-c7ac8960258f)
ref: [allow multiple states](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6736/commits/cdb884e23199d1deb9022953d6a5ec83be25c895)